### PR TITLE
github: write build summary

### PR DESCRIPTION
**Pull request by crazy-max**
* ~needs https://github.com/docker/actions-toolkit/pull/307~

### CODE REVIEW

File: src/buildx/history.ts [hunk 1]
```diff
@@ -35,7 +35,7 @@ export interface HistoryOpts {
 export class History {
   private readonly buildx: Buildx;
 
-  private static readonly EXPORT_TOOL_IMAGE: string = 'docker.io/dockereng/export-build:latest';
+  private static readonly EXPORT_TOOL_IMAGE: string = 'docker.io/crazymax/export-build:latest';
```
Comment (crazy-max) [inline src/buildx/history.ts:38]: revert this change when summary is handled in export-build tool

File: src/github.ts [hunk 2]
```diff
@@ -190,4 +195,84 @@ export class GitHub {
...
+        .addRaw(addLink('Learn more', 'https://docs.docker.com/desktop/use-desktop/builds/#import-build-records'))
+      .addRaw('</p>')
+      .addRaw(`<p>`)
+        .addRaw(`:arrow_down: ${addLink(`<strong>${opts.uploadRes.filename}</strong>`, opts.uploadRes.url)} (${Util.formatFileSize(opts.uploadRes.size)})`)
+        .addBreak()
+        .addRaw(`This file includes <strong>${refsSize} build record${refsSize > 1 ? 's' : ''}</strong>.`)
+      .addRaw(`</p>`)
+      .addRaw(`<p>`)
+        .addRaw(`Was this useful? `)
+        .addRaw(addLink('Let us know', 'https://docs.docker.com/feedback/desktop-build'))
```
Comment (crazy-max) [inline src/github.ts:224]: This link needs to be changed so it redirects to the GHA summary feedback survey (cc @colinhemmings @dvdksn)

File: src/github.ts [hunk 3]
```diff
@@ -190,4 +195,84 @@ export class GitHub {
...
+    const refsSize = Object.keys(opts.exportRes.refs).length;
+
+    // prettier-ignore
+    const sum = core.summary
+      .addHeading('Docker Build summary', 1)
+      .addRaw(`<p>`)
+        .addRaw(`For a detailed look at the build, download the following <code>.dockerbuild</code> file and import it into Docker Desktop's Build views. `)
+        .addBreak()
+        .addRaw(`The build file provides insights of your build, such as timing, dependencies, results, logs, traces, and more. `)
+        .addRaw(addLink('Learn more', 'https://docs.docker.com/desktop/use-desktop/builds/#import-build-records'))
```
Comment (crazy-max) [inline src/github.ts:215]: @dvdksn This anchor doesn't exist yet but it should instruct users of how to import builds in Docker Desktop. I will open an issue related to this shortly.

File: src/github.ts [hunk 4]
```diff
@@ -190,4 +195,84 @@ export class GitHub {
...
+      return `<a href="${url}">${text}</a>` + (addEOL ? os.EOL : '');
+    };
+
+    const refsSize = Object.keys(opts.exportRes.refs).length;
+
+    // prettier-ignore
+    const sum = core.summary
+      .addHeading('Docker Build summary', 1)
+      .addRaw(`<p>`)
+        .addRaw(`For a detailed look at the build, download the following <code>.dockerbuild</code> file and import it into Docker Desktop's Build views. `)
```
[the remark this answers is not in the document]
Comment (crazy-max) [reply src/github.ts:212]: yes indeed it was swapped, should called "Builds view" to match:

image

File: src/github.ts [hunk 5]
```diff
@@ -190,4 +195,84 @@ export class GitHub {
...
+        .addRaw(`The build file provides insights into your build, such as timing, dependencies, results, logs, traces, and more. `)
+        .addRaw(addLink('Learn more', 'https://docs.docker.com/desktop/use-desktop/builds/#import-build-records'))
+      .addRaw('</p>')
+      .addRaw(`<p>`)
+        .addRaw(`:arrow_down: ${addLink(`<strong>${opts.uploadRes.filename}</strong>`, opts.uploadRes.url)} (${Util.formatFileSize(opts.uploadRes.size)})`)
+        .addBreak()
+        .addRaw(`This file includes <strong>${refsSize} build record${refsSize > 1 ? 's' : ''}</strong>.`)
+      .addRaw(`</p>`)
+      .addRaw(`<p>`)
+        .addRaw(`Was this useful? `)
```
Comment (dvdksn) [inline src/github.ts:223]: I'm not sure about the past tense here - this text appears before the summary. Maybe "Find this useful?" could work?
```suggestion
        .addRaw(`Find this useful? `)
```

File: src/github.ts [hunk 6]
```diff
@@ -190,4 +195,84 @@ export class GitHub {
...
+
+    const refsSize = Object.keys(opts.exportRes.refs).length;
+
+    // prettier-ignore
+    const sum = core.summary
+      .addHeading('Docker Build summary', 1)
+      .addRaw(`<p>`)
+        .addRaw(`For a detailed look at the build, download the following <code>.dockerbuild</code> file and import it into Docker Desktop's Builds view. `)
+        .addBreak()
+        .addRaw(`The build file provides insights into your build, such as timing, dependencies, results, logs, traces, and more. `)
```
Comment (dvdksn) [inline src/github.ts:214]: question: (I might be totally out of line here / missing some context) Do we need to emphasize the `.dockerbuild` extension at all? It seems like the downloaded file is a `.zip` archive image layout. Could we maybe just call it a "build record"? Something like:

```suggestion
        .addRaw(`This is a summary of the build job. For the full details about your build, download the build record and import it into Docker Desktop's Builds view.`)
        .addBreak()
        .addRaw(`The build record includes data such as timing, dependencies, results, logs, traces, and more.`)
```

File: src/github.ts (hunk 5 above)
Comment (crazy-max) [reply src/github.ts:223]: Yes agree!

File: src/github.ts (hunk 6 above)
Comment (crazy-max) [reply src/github.ts:214]: There can be multiple build records within a `.dockerbuild` file (with bake): 

image

`.dockerbuild` is more like a bundle having the build record and associated blob suc has traces, provenance, ...

Comment (dvdksn) [reply src/github.ts:214]: Maybe we can call the downloadable artifact a "build record archive" then, and avoid calling it `.dockerbuild` file (since that is not very descriptive). A build record archive contains one or more build records, each of which contains the logs, traces, provenance, etc for a single build invocation

Comment (crazy-max) [reply src/github.ts]: @dvdksn Updated: https://github.com/docker/actions-toolkit/actions/runs/9094081924?pr=324#summary-24994391676

image

File: src/github.ts [hunk 7]
```diff
@@ -190,4 +195,84 @@ export class GitHub {
...
+
+    const refsSize = Object.keys(opts.exportRes.refs).length;
+
+    // prettier-ignore
+    const sum = core.summary
+      .addHeading('Docker Build summary', 1)
+      .addRaw(`<p>`)
+        .addRaw(`For a detailed look at the build, download the following build record archive and import it into Docker Desktop's Builds view. `)
+        .addBreak()
+        .addRaw(`The build file provides insights into your build, such as timing, dependencies, results, logs, traces, and more. `)
```
Comment (dvdksn) [inline src/github.ts:214]: Re-reading this, and thinking that "a file provides insights" can be improved, suggestion:

```suggestion
        .addRaw(`Build records include details such as timing, dependencies, results, logs, traces, and other information about a build. `)
```

Comment (crazy-max) [reply src/github.ts]: https://github.com/docker/actions-toolkit/actions/runs/9095301220?pr=324#summary-24998248180

image

Comment (dvdksn) [reply src/github.ts]: LGTM

File: src/github.ts [hunk 8]
```diff
@@ -190,4 +195,84 @@ export class GitHub {
...
+        .addRaw(addLink('Learn more', 'https://docs.docker.com/desktop/use-desktop/builds/#import-build-records'))
+      .addRaw('</p>')
+      .addRaw(`<p>`)
+        .addRaw(`:arrow_down: ${addLink(`<strong>${opts.uploadRes.filename}</strong>`, opts.uploadRes.url)} (${Util.formatFileSize(opts.uploadRes.size)})`)
+        .addBreak()
+        .addRaw(`This file includes <strong>${refsSize} build record${refsSize > 1 ? 's' : ''}</strong>.`)
+      .addRaw(`</p>`)
+      .addRaw(`<p>`)
+        .addRaw(`Find this useful? `)
+        .addRaw(addLink('Let us know', 'https://docs.docker.com/feedback/gha-build-summary'))
```
Comment (crazy-max) [inline src/github.ts:224]: This redirect will be set in https://github.com/docker/docs/pull/20043

File: src/github.ts [hunk 9]
```diff
@@ -190,4 +195,84 @@ export class GitHub {
...
+    const refsSize = Object.keys(opts.exportRes.refs).length;
+
+    // prettier-ignore
+    const sum = core.summary
+      .addHeading('Docker Build summary', 1)
+      .addRaw(`<p>`)
+        .addRaw(`For a detailed look at the build, download the following build record archive and import it into Docker Desktop's Builds view. `)
+        .addBreak()
+        .addRaw(`Build records include details such as timing, dependencies, results, logs, traces, and other information about a build. `)
+        .addRaw(addLink('Learn more', 'https://docs.docker.com/build/ci/github-actions/build-summary'))
```
[the remark this answers is not in the document]
Comment (dvdksn) [reply src/github.ts:215]: Added one a go-redirect here: https://github.com/docker/docs/pull/20057/commits/61e94bd803652cbb8cc05a25c3746310da12c3ab

[the remark this answers is not in the document]
Comment (crazy-max) [reply src/github.ts]: Updated to use the `/go` link thanks!

**Comment by crazy-max** [comment]
Taking this one in to unblock downstream PRs
